packages/amplify-util-mock/src/utils/opensearch.ts

```diff
@@ -0,0 +1,57 @@
+import { $TSObject, AMPLIFY_SUPPORT_DOCS, AmplifyError } from 'amplify-cli-core';
+import fetch from 'node-fetch';
+
+type OpensearchQueryConfig = {
+  path: string,
+  params: {
+    body: {
+      size: number,
+      sort: $TSObject[],
+      version: boolean,
+      query: $TSObject,
+      aggs: $TSObject,
+      from?: number
+    }
+  }
+}
+
+type OpensearchQueryResult = {
+  hits: {
+    hits: $TSObject,
+    total: {
+      value: number
+    }
+  },
+  aggregations: $TSObject,
+
+}
+
+export const querySearchable = async (endpoint: string, searchConfig: OpensearchQueryConfig): Promise<OpensearchQueryResult> => {
+  if (!endpoint) {
+    throw new AmplifyError('MockProcessError', {
+      message: 'The local opensearch endpoint is not found',
+      link: AMPLIFY_SUPPORT_DOCS.CLI_GRAPHQL_TROUBLESHOOTING.url
+    });
+  }
+
+  try {
+    searchConfig = searchConfig as OpensearchQueryConfig;
+  }
+  catch(e) {
+    throw new AmplifyError('MockProcessError', {
+      message: 'Given search query configuration is not valid',
+      link: AMPLIFY_SUPPORT_DOCS.CLI_GRAPHQL_TROUBLESHOOTING.url
+    });
```

Comment:
After diving into the code I think these are Faults, sorry about the mistakes @phani-srikar